### PR TITLE
Feature/adapter

### PR DESCRIPTION
**Pull request by kovpaka**
solved the adapter pattern task (I hope)

### CODE REVIEW

File: adapter/Adapter.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    class Adapter : ITarget
+    {
+        private readonly BookAnalyzer bookAnalyzer;
+
+        public Adapter(BookAnalyzer bookAnalyzer)
+        {
+            this.bookAnalyzer = bookAnalyzer;
+        }
+
+        public string getBookInJson(string xml)
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply adapter/Adapter.cs]: Fixed

File: adapter/Adapter.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Xml;
+
+namespace adapter
+{
+    class Adapter : ITarget
+    {
+        public Adapter() { }
+        public List<Book> getOldestBook(string xml)
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply adapter/Adapter.cs]: It's my bad, fixed

File: adapter/Adapter.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        {
+            XmlDocument doc = new XmlDocument();
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply adapter/Adapter.cs]: Fixed.

File: adapter/BookAnalyzer.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace adapter
+{
+    class BookAnalyzer
+    { 
+        public Book getOldestBook(List<Book> bookslist)
+        {
+            var result = from books in bookslist
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply adapter/BookAnalyzer.cs]: Got it, fixed.

File: adapter/Program.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+namespace adapter
+{
+    class Program
+    {
+        static void Main()
+        {
+            Library library = new Library();
+            XmlDocument xml = library.getBooksXML();
+            Adapter adapter = new Adapter();
+            List<Book> s = adapter.getOldestBook(xml.OuterXml);
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply adapter/Program.cs]: Changed the structure of the calling method.

File: adapter/Adapter.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        private readonly Library library;
+        public Adapter(Library library) 
+        {
+            this.library = library;
+        }
+
+        public List<Book> GetListOfBooksInJson()
+        {
+            var books = new List<Book>();
+            try
```
Comment (siarheikarabko) [inline adapter/Adapter.cs:20]: Let's discuss this try-catch block.

**Pull request merged**